src/main/webapp/3Dscript.js

```diff
@@ -45,146 +76,272 @@ function initThreeJs() {
+ * request and significantly cut down rendering time with
  * geometry instancing.
+ * @param {Object} center An LLA coordinate used as the center
+ *     of the 2D map.
 */
-function addMap() {
+function addMap(center) {
   // Creates the 2D map.
-  let loader = new THREE.TextureLoader();
-  let material = new THREE.MeshLambertMaterial({ 
+  const loader = new THREE.TextureLoader();
+  const material = new THREE.MeshLambertMaterial({
     map: loader.load(
-      'https://maps.googleapis.com/maps/api/staticmap' +
-      '?format=png&center=' + pose[0].lat + ',' + pose[0].lng + 
-      '&zoom=18&size=500x500&key=' + apiKey)
-    });
-  let geometry = new THREE.PlaneGeometry(50,50);
+        'https://maps.googleapis.com/maps/api/staticmap' +
+      '?format=png&center=' + center.latitude + ',' + center.longitude +
+      '&zoom=18&size=500x500&key=' + apiKey),
+  });
+  const geometry = new THREE.PlaneGeometry(200, 200);
   const map = new THREE.Mesh(geometry, material);
-  map.position.set(0,-4,0);
-  map.rotation.x = -Math.PI / 2;
+  map.position.set(0, 0, -4);
   scene.add(map);
 
   // Add the light to the scene.
-  let light = new THREE.PointLight(0xffffff, 1, 0 );
-  light.position.set(0, 100, 0);
-  scene.add(light)
-  plotTrajectory();
-  
-  // Adds pose data to the scene as one Geometry instance.
+  const light = new THREE.PointLight(0xffffff, 1, 0 );
+  light.position.set(0, 0, 1000);
+  scene.add(light);
+}
+
+
+/**
+ * Creates all the children for each axis of orientation. Each object
+ * is a geometry instance, initializing the object with one geometry
+ * call rather than the number of data points. (O(1) vs. O(N))
+ * @return {object} A Point object holding a cylinder for each axis.
+ */
+function createInstances() {
+  // Blue corresponds to the Z axis.
+  let material = new THREE.MeshBasicMaterial({color: 'blue'});
+  const geometry = new THREE.CylinderBufferGeometry(.005, .005, .1);
+  const zCylinder = new THREE.InstancedMesh(geometry, material, pose.length);
+  zCylinder.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(zCylinder);
+
+  // Green corresponds to the Y axis.
+  material = new THREE.MeshBasicMaterial({color: 'green'});
+  const yCylinder = new THREE.InstancedMesh(geometry, material, pose.length);
+  yCylinder.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(yCylinder);
+
+  // Red corresponds to the X axis.
   material = new THREE.MeshBasicMaterial({color: 'red'});
-  geometry = new THREE.CylinderBufferGeometry(.005, .005, .1);
-  orientation = new THREE.InstancedMesh(geometry, material, pose.length);
-  orientation.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
-  scene.add(orientation);
-  plotOrientation();
+  const xCylinder = new THREE.InstancedMesh(geometry, material, pose.length);
+  xCylinder.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(xCylinder);
+
+  return new Point(xCylinder, yCylinder, zCylinder);
```

Comment:
This seems like an abuse of `Point`. I wouldn't expect a point to have anything other than scalar components.